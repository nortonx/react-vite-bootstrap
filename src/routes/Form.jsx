import { useState } from "react";
import "../scss/Form.scss";
import Input from "../components/Input";
import Button from "../components/Button";
const Form = () => {
  const [email, setEmail] = useState("");
  return(
    <div className="container form" data-testid="form">
      <div className="row">
        <div className="col-md-6">
          <Input
            label="E-mail:"
            type="email"
            value={email}
            onChange={(e) => setEmail(e.target.value)}
          />
          <Button
            label="Testing button"
            
          />
        </div>
      </div>
    </div>
  );
};

export default Form;