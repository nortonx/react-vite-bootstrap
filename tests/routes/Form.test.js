import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";

import Form from "../../src/routes/Form";

describe("Form route", () => {
  it("Should render the component and update snapshot", async () => {
    render(<Form />);
    const element = await screen.findByTestId("form");
    expect(element).toBeInTheDocument();
    const form = renderer.create(<Form />);
    expect(form).toMatchSnapshot();
  });
})