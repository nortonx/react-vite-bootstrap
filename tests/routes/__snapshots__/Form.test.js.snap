// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form route Should render the component and update snapshot 1`] = `
<div
  className="container form"
  data-testid="form"
>
  <div
    className="row"
  >
    <div
      className="col-md-6"
    >
      <code />
    </div>
    <div
      className="col-md-6"
    >
      <div
        className="col-sm-6"
      >
        <label
          className="form-label"
          htmlFor="userEmail"
        >
          E-mail:
        </label>
        <input
          className="form-control input mb-3"
          data-testid="input"
          id="userEmail"
          onChange={[Function]}
          type="email"
          value=""
        />
      </div>
      <button
        className="btn btn-primary"
        data-testid="button"
      >
        Testing button
      </button>
    </div>
  </div>
</div>
`;
